### PR TITLE
k6 load test scripts

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Adds scripts to:

- install [k6](https://k6.io/) mac and linux binaries
- run a k6 load test script for articles

## Why?

Convenience to help run k6 load tests rather than manually installing each time.

## Screenshots

On a t4g.micro instance:

```
          /\      |‾‾| /‾‾/   /‾‾/
     /\  /  \     |  |/  /   /  /
    /  \/    \    |     (   /   ‾‾\
   /          \   |  |\  \ |  (‾)  |
  / __________ \  |__| \__\ \_____/ .io

  execution: local
     script: k6.js
     output: -

  scenarios: (100.00%) 1 scenario, 10 max VUs, 1m30s max duration (incl. graceful stop):
           * default: 10 looping VUs for 1m0s (gracefulStop: 30s)


     ✓ is status 200

     checks.........................: 100.00% ✓ 1310     ✗ 0
     data_received..................: 399 MB  6.6 MB/s
     data_sent......................: 87 MB   1.4 MB/s
     http_req_blocked...............: avg=116.81µs min=4.87µs   med=6.3µs    max=32.62ms p(90)=7.37µs   p(95)=10.61µs
     http_req_connecting............: avg=90.82µs  min=0s       med=0s       max=19.54ms p(90)=0s       p(95)=0s
     http_req_duration..............: avg=448.74ms min=314.14ms med=408.14ms max=1.66s   p(90)=755.9ms  p(95)=809.94ms
       { expected_response:true }...: avg=448.74ms min=314.14ms med=408.14ms max=1.66s   p(90)=755.9ms  p(95)=809.94ms
     http_req_failed................: 0.00%   ✓ 0        ✗ 1310
     http_req_receiving.............: avg=257.42µs min=117.77µs med=183.76µs max=12.26ms p(90)=383.48µs p(95)=413.88µs
     http_req_sending...............: avg=448.1µs  min=100.74µs med=160.73µs max=12.51ms p(90)=345.73µs p(95)=1.86ms
     http_req_tls_handshaking.......: avg=0s       min=0s       med=0s       max=0s      p(90)=0s       p(95)=0s
     http_req_waiting...............: avg=448.03ms min=313.8ms  med=407.6ms  max=1.66s   p(90)=755.59ms p(95)=809.52ms
     http_reqs......................: 1310    21.68247/s
     iteration_duration.............: avg=459.41ms min=330.21ms med=418.98ms max=1.67s   p(90)=766.34ms p(95)=819.45ms
     iterations.....................: 1310    21.68247/s
     vus............................: 10      min=10     max=10
     vus_max........................: 10      min=10     max=10
```

### CODE REVIEW

File: dotcom-rendering/scripts/perf/k6/k6-install.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+tar -xvf $LINUX_VERSION
+rm $LINUX_VERSION
+echo "Downloading macos-arm64 binary"
+wget https://github.com/grafana/k6/releases/download/$VERSION/$MAC_VERSION
+unzip $MAC_VERSION
+rm $MAC_VERSION
+echo "to run a test:"
+echo "./k6-$VERSION-macos-arm64/k6 run k6.js"
+echo "or"
+echo "./k6-$VERSION-linux-arm64/k6 run k6.js"
```
Comment (mxdvl) [inline dotcom-rendering/scripts/perf/k6/k6-install.sh:16]: Do we need to download both versions on all machines?

https://stackoverflow.com/questions/3466166/how-to-check-if-running-in-cygwin-mac-or-linux

File: dotcom-rendering/scripts/perf/k6/k6.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const jsonPayload = new SharedArray('jsonPayload', function () {
+	const f = JSON.parse(open('./article-nier-automata.json'));
+	return [f];
+});
+
+export const options = {
+	vus: 10,
+	duration: '60s',
+};
```
Comment (mxdvl) [inline dotcom-rendering/scripts/perf/k6/k6.js:16]: Could this be checked against a `k6` config type?

**Review by mxdvl — COMMENTED** [review]
Load testing is good.

Would this eventually run automatically, or is this mostly about capturing how to run this test on real AWS instances?

**Comment by arelra** [comment]
> Load testing is good.
> 
> Would this eventually run automatically, or is this mostly about capturing how to run this test on real AWS instances?

Yes I was planning to add to CI but I need to work out the performance profile of the github runners as they are not directly comparable to our prod instances.

For now its more for the ease of testing on actual instances.

**Pull request merged**